sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -170,6 +170,23 @@ case class CatalogTablePartition(
   }
 }
 
+/**
+ * A container for clustering information.
+ *
+ * @param columnNames the names of the columns used for clustering.
+ */
+case class ClusterBySpec(columnNames: Seq[UnresolvedAttribute]) {
+  override def toString: String = columnNames.map(_.name).mkString(",")
+
+  lazy val toDDL: String = if (columnNames.nonEmpty) s"CLUSTER BY ($toString)" else ""
+}
+
+object ClusterBySpec {
+  def apply(columns: String): ClusterBySpec = columns match {
```

Comment:
this looks weird, where do we use it?

Reply:
this is used for parsing property value back to ClusterBySpec. I renamed this to `fromProperty`.